### ISSUE TITLE
Removing renderer

### ISSUE THREAD

**Issue by carvan** [philippe44/AirConnect#514]
Hi there

first off thanks a lot for your work. Could you please help me understand why this is happening?

Jan 03 08:01:05 airconnect aircast-linux-x86_64-static[127]: [08:01:05.955] AddCastDevice:650 [0x192ce20]: creating MAC
Jan 03 08:01:05 airconnect aircast-linux-x86_64-static[127]: [08:01:05.955] AddCastDevice:664 [0x192ce20]: adding renderer (Cast Audio - 192.168.6.83:8009) with mac CCCC3C2E14D0
Jan 03 08:02:34 airconnect aircast-linux-x86_64-static[127]: [08:02:34.477] AddCastDevice:650 [0x192d7b8]: creating MAC
Jan 03 08:02:34 airconnect aircast-linux-x86_64-static[127]: [08:02:34.479] AddCastDevice:664 [0x192d7b8]: adding renderer (Cast Letto - 192.168.5.85:8009) with mac CCCC85B97153
Jan 03 08:17:40 airconnect aircast-linux-x86_64-static[127]: [08:17:40.131] UpdateDevices:397 [0x192ce20]: removing renderer (Cast Audio+) -536835552
Jan 03 08:17:40 airconnect aircast-linux-x86_64-static[127]: [08:17:40.161] DeleteCastDevice:379 [0x192ce20]: Cast device stopped

it looks like that particular device (an "old" Google Cast Audio) is getting removed for apparently no reason. If I restart the service it then works fine again. Any hints about what could be causing this?

**Comment by philippe44** [comment]
Devices get removed because they stop updating their mDNS record. They have to but there is a feature in networking named IGMP snooping that is supposed to help reducing traffic but is quite often a mess.

The idea is that when you are interested in listening to a multicast group, you subscribe and then routers will direct such multicast to your port/branch of network. IGMP snooping allows that for simple switch, there are many good explanation on the internet https://www.cloudflare.com/learning/network-layer/what-is-igmp/. 

Well, the problem is that IGMP snooping does seem to not work on many switches and they end up blocking traffic and so I don't see them after a while and remove them. Which version do you use? In 1.6.2 I've added an extra feature that pings a missing player "to be sure"

**Comment by carvan** [comment]
Hello and thanks a lot for your reply and explanations. I run the 1.6.0, I'll update it and see what happens then.

**Comment by carvan** [comment]
Hi
it looks like the issue isn't gone:
![image](https://github.com/philippe44/AirConnect/assets/11394057/5e68d1fa-1145-4ae8-be93-799b398eea1d)

In the meanwhile, I noticed that I do not have the IMGP snooping enabled on all my network devices (I have a "non standard" network config with 1 router, 1 access point and 1 switch), could this be the issue as well? Or at least part of it?

**Comment by philippe44** [comment]
Yes, it's very probably part of it, the fact that one of your equipment is dropping multicast frames. I think 1.6.1 and above  should solve that issue

**Comment by carvan** [comment]
Ok I'll make sure to have that option enabled on all my network equipment and see what happens
